tabular/src/autogluon/tabular/task/tabular_prediction/predictor.py

```diff
@@ -598,34 +599,38 @@ def feature_importance(self, dataset=None, model=None, features=None, feature_st
             If `subsample_size=None` or `dataset` contains fewer than `subsample_size` rows, all rows will be used during computation.
             Larger values increase the accuracy of the feature importance scores.
             Runtime linearly scales with `subsample_size`.
+        time_limit : float, default = None
+            Time in seconds to limit the calculation of feature importance.
+            If None, feature importance will calculate without early stopping.
+            A minimum of 1 full shuffle set will always be evaluated. If a shuffle set evaluation takes longer than `time_limit`, the method will take the length of a shuffle set evaluation to return regardless of the `time_limit`.
+        num_shuffle_sets : int, default = None
+            The number of different permutation shuffles of the data that are evaluated.
+            Larger values will increase the quality of the importance evaluation.
+            It is generally recommended to increase `subsample_size` before increasing `num_shuffle_sets`.
+            Defaults to 1 if `time_limit` is None or 10 if `time_limit` is specified.
+            Runtime linearly scales with `num_shuffle_sets`.
         silent : bool, default = False
             Whether to suppress logging output
 
         Returns
         -------
-        Pandas `pandas.Series` of feature importance scores.
-
+        Pandas `pandas.DataFrame` of feature importance scores with 3 columns:
+            index: The feature name.
+            'importance': The feature importance score.
+            'stddev': The standard deviation of the feature importance score. If NaN, then not enough num_shuffle_sets were used to calculate a variance.
+            'z_score': The z-score of the feature importance score. Equivalent to 'importance' / 'stddev'.
+                A z-score of +4 or higher indicates that the feature is almost certainly useful and should be kept.
+                A z-score of +2 indicates that the feature has a 97.5% chance of improving model quality when present.
+                A z-score that is 0 or negative indicates that the feature can likely be dropped without negative impact to model quality.
+                A z-score of NaN indicates that the feature's stddev was NaN or that the model predictions were never impacted by the feature (importance 0 and stddev 0). This indicates that the feature can safely be dropped.
+                A z-score of +inf or -inf indicates that `subsample_size` and/or `num_shuffle_sets` were too small to calculate variance for the feature (non-zero importance with zero stddev).
```

Comment:
Would finally also describe how to construct say a 95% confidence interval for a particular feature's importance. And also how the user can shrink this interval by changing time_limits or other args.

Reply:
Added a bunch of improvements regarding this, was actually calculating z-score wrong (and we shouldn't even be using z-score, should actually be using t-statistic which we are now).

I added p_score and 99% confidence interval to output along with all the information needed to calculate arbitrary confidence intervals (mean, stddev, n).